pkg/policies/loader.go

```diff
@@ -40,62 +43,86 @@ const (
 
 // Loader defines the interface for policy loaders from contract attachments
 type Loader interface {
-	Load(context.Context, *v1.PolicyAttachment) (*v1.Policy, error)
+	Load(context.Context, *v1.PolicyAttachment) (*v1.Policy, *v12.ResourceDescriptor, error)
 }
 
 // EmbeddedLoader returns embedded policies
 type EmbeddedLoader struct{}
 
-func (e *EmbeddedLoader) Load(_ context.Context, attachment *v1.PolicyAttachment) (*v1.Policy, error) {
-	return attachment.GetEmbedded(), nil
+const chainloopLoaderPrefix = "chainloop://"
+
+func (e *EmbeddedLoader) Load(_ context.Context, attachment *v1.PolicyAttachment) (*v1.Policy, *v12.ResourceDescriptor, error) {
+	return attachment.GetEmbedded(), nil, nil
 }
 
 // FileLoader loader loads policies from filesystem and HTTPS references using Cosign's blob package
 type FileLoader struct{}
 
-func (l *FileLoader) Load(_ context.Context, attachment *v1.PolicyAttachment) (*v1.Policy, error) {
+func (l *FileLoader) Load(_ context.Context, attachment *v1.PolicyAttachment) (*v1.Policy, *v12.ResourceDescriptor, error) {
 	var (
-		rawData []byte
-		err     error
+		raw []byte
+		err error
 	)
 
 	ref := attachment.GetRef()
 	filePath, err := ensureScheme(ref, fileScheme)
 	if err != nil {
-		return nil, err
+		return nil, nil, fmt.Errorf("invalid policy reference %q", ref)
```

Comment:
For these cases, where we just use auxiliary functions in the same package, I think wrapping the error is too much duplicity (see the error from `ensureScheme`, it's exactly the same message).

Reply:
ok